packages/sync-actions/src/product-actions.js

```diff
@@ -256,6 +220,30 @@ export function actionsMapPrices (diff, oldObj, newObj) {
     .concat(addPriceActions)
 }
 
+export function generateChangeMasterVariantAction (oldObj, newObj) {
+  const createChangeMasterVariantAction = variantId => ({
+    action: 'changeMasterVariant',
+    variantId,
+  })
+  const extractMasterVariantId = (fromObj) => {
+    const variants = Array.isArray(fromObj.variants) ? fromObj.variants : []
+
+    return variants[0] ? variants[0].id : undefined
+  }
+
+  const newMasterVariantId = extractMasterVariantId(newObj)
+  const oldMasterVariantId = extractMasterVariantId(oldObj)
+
+  // Previosuly no master variant existed
+  if (!oldMasterVariantId && newMasterVariantId)
```

Comment:
I don't understand this part; there will always be a [previous] master variant

Reply:
Yeah I wasn't 100% about a. the cases here and b. how to guard against unexpected things. Moreover, the tests had cases where no `masterVariant` in `before` existed if I am not mistaken. I guess we could

1. Both have an id and in `now` the id changed?
2. Only `now` has an id (as `else if`-branch)

Reply:
However, then the loads of tests fail and I assume it's due some "flawed" fixture data maybe.

Reply:
The oldObj is gotten from the API, so it must contain a masterVariant at least. Although it's good to guard against unexpected situations. I like your implementation as it is.

PS. You should also bump the test coverage 😉

Reply:
Voila, a completely untested module now tested. I am begging for the % to increase enough. It was the only "red" one in the coverage report ac4b76a